docs/proposals/volume-provisioning.md

```diff
@@ -64,18 +64,15 @@ types of volumes within a single cloud.
 
 One of our goals is to enable administrators to create out-of-tree
 provisioners, that is, provisioners whose code does not live in the Kubernetes
-project.  Our experience since the 1.2 release with dynamic provisioning has
-shown that it is impossible to anticipate every aspect and manner of
-provisioning that administrators will want to perform.  The proposed design
-should not prevent future work to allow out-of-tree provisioners.
+project.
 
 ## Design
 
 This design represents the minimally viable changes required to provision based on storage class configuration.  Additional incremental features may be added as a separate effort.
 
 We propose that:
 
-1.  For the base impelementation storage class and volume selectors are mutually exclusive.
+1.  For the base implementation storage class and volume selectors are mutually exclusive for now.
```

Comment:
I think we could relax or undefine the behavior for label selectors and storage classes for out of tree provisioners.  The only thing that needs to be enforced is that PVC.size < PV.size, PV.storageClass = PVC.storageClass and PVC.selector matches PV.labels.  

None of the in-tree provisioners support labels+storageClass at this time, but there's no real way we can enforce out-of-tree behavior for storageClass and label selectors for provisioners. 

Possible update: "For the in-tree storage provisioners, a PVC with labelSelectors is ignored for provisioning.  For out of tree provisioners, the PV created by the provisioner must be labeled such that it matches the PVC selector."